### ISSUE TITLE
[Accessibility], WCAG 2.4.3. Keyboard focus isn't taken to the Suggest a new concept -modals

### ISSUE THREAD

**Issue by Eiskore** [NatLibFi/Skosmos#1988]
## URL address of the page where you encountered the problem
https://test.dev.finto.fi/yso/en/ and YSO consept pages

## Description of the problem
When the suggestion form is opened, the keyboard focus is not moved to the opened modal, but the focus remains on the page below. On the front page of YSO vocabulary, the modal is next in focus order after triggering button, so keyboard and screen reader users can still find their way to the modal fairly easily. 
But the modal opened on concept pages is not as easily accessible, because the modal is the last element of the page in the focus order (although at the moment keyboard users cannot open modals from concept pages at all). On concept pages the next tab press moves the focus back to the tabs on the left. The keyboard focus can be moved to the modal at the end of the page. 

**Fix**: The modal should be made into a closed environment where the keyboard focus is taken directly after the modal has opened. In such a modal, the focus should be directed to the beginning of the modal (the heading or close button). Also make sure that the keyboard focus (focustrap) and screen reader cursor (e.g. inert or aria-hidden="true") do not accidentally escape from the modal to the page below and blocking background content.
Also make sure that when the modal is closed, the focus returns to the point where it wae when the modal opened.

**How to test:** 
Precondition: Suggest a change to a concept and Suggest a new concept for the vocabulary -buttons are fixed and they are keyboard focusable.

Turn on the screen reader. Navigate with tab key to either button and press space:

**Expected result:** A modal dialog opens and keyboard focus is taken into the modal, either to the heading or to the close button. Keyboard focus (tab) cannot escape from the modal to the page below. When the last focusable element of the modal is focused and tab is pressed again, keyboard focus is directed back to the modal’s up corner, close button. 
Next browse the modal content with screen reader’s browse mode (up and down arrow keys). When screen reader focus is at the end of the modal, press down arrow. Screen reader repeats the last element at the modal. It is the last content screen reader can find (Screen reader focus is not shifted to the page below the modal).

Severity: Serious
Priority: High

## Additional information (e.g. screenshots) about the problem

<img width="1004" height="659" alt="Image" src="https://github.com/user-attachments/assets/e756c666-98f5-4007-9eb6-715dac5b220c" />

## The browser you used when the problem appeared
Chrome, Firefox, Edge
